engine/core/src/main/java/org/datacleaner/configuration/RemoteServerConfigurationImpl.java

```diff
@@ -22,19 +22,40 @@
 import java.util.ArrayList;
 import java.util.List;
 
+import org.apache.commons.collections.list.UnmodifiableList;
+
 /**
  * Implementation of {@link RemoteServerConfiguration}.
  */
 public class RemoteServerConfigurationImpl implements RemoteServerConfiguration {
 
-    private final List<RemoteServerData> remoteServerDataList;
+    protected final List<RemoteServerData> remoteServerDataList;
 
-    public RemoteServerConfigurationImpl() {
-        remoteServerDataList = new ArrayList<>();
+    public RemoteServerConfigurationImpl(List<RemoteServerData> serverData) {
+        remoteServerDataList = new ArrayList<>(serverData);
     }
-
     @Override
     public List<RemoteServerData> getServerList() {
-        return remoteServerDataList;
+        return UnmodifiableList.decorate(remoteServerDataList);
```

Comment:
Normally I would use ```Collections.unmodifyableList(...)``` here. I think the apache-commons variant is there only for JDK4 users.